zstash/utils.py

```diff
@@ -5,10 +5,141 @@
+        url: ParseResult = urlparse(hpss_path)
+        if url.scheme != "globus":
+            raise ValueError(f"Invalid Globus hpss_path={hpss_path}")
+        self.hpss_path: str = hpss_path
+        self.url: ParseResult = url
+
+        # Set in globus.globus_activate
+        self.remote_endpoint: Optional[str] = None
+        self.local_endpoint: Optional[str] = None
+        self.transfer_client: Optional[TransferClient] = None
+
+        # Set in globus.globus_transfer
+        self.archive_directory_listing: Optional[IterableTransferResponse] = None
+        self.transfer_data: Optional[TransferData] = None
+        self.task_id = None
+        self.tarfiles_pushed: int = 0
+
+
+# Class to hold configuration, as it appears in the database
+class Config(object):
+    path: Optional[str] = None
+    hpss: Optional[str] = None
+    maxsize: Optional[int] = None
+
+
+class CommandInfo(object):
+
+    def __init__(self, command_name: str):
+        self.command_name: str = command_name
+        self.dir_called_from: str = os.getcwd()
+        self.cache_dir: str = "zstash"  # # Default sub-directory to hold cache
+        self.config: Config = Config()
+        self.keep: bool = False  # Defaults to False
+        self.prev_transfers: List[str] = []
+        self.curr_transfers: List[str] = []
+        # Use set_dir_to_archive:
+        self.dir_to_archive_relative: Optional[str] = None
+        # Use set_hpss_parameters:
+        self.hpss_type: Optional[HPSSType] = None
+        self.globus_info: Optional[GlobusInfo] = None
+
+    def set_dir_to_archive(self, path: str):
+        abs_path = os.path.abspath(path)
+        if abs_path is not None:
+            self.config.path = abs_path
+            self.dir_to_archive_relative = path
+        else:
+            raise ValueError(f"Invalid path={path}")
+
+    def set_and_scale_maxsize(self, maxsize):
+        self.config.maxsize = int(1024 * 1024 * 1024 * maxsize)
+
+    def validate_maxsize(self):
+        if self.config.maxsize is not None:
+            self.config.maxsize = int(self.config.maxsize)
+        else:
+            raise ValueError("config.maxsize is undefined")
+
+    def set_hpss_parameters(self, hpss_path: str, null_hpss_allowed=False):
+        self.config.hpss = hpss_path
+        if hpss_path == "none":
+            self.hpss_type = HPSSType.NO_HPSS
+        elif hpss_path is not None:
+            url = urlparse(hpss_path)
+            if url.scheme == "globus":
+                self.hpss_type = HPSSType.GLOBUS
+                self.globus_info = GlobusInfo(hpss_path)
+                globus_cfg: str = os.path.expanduser("~/.globus-native-apps.cfg")
+                logger.info(f"Checking if {globus_cfg} exists")
+                if os.path.exists(globus_cfg):
+                    logger.info(f"Removing {globus_cfg}")
+                    # Otherwise, may cause "Token is not active" TransferAPIError
+                    os.remove(globus_cfg)
```

Comment:
I was able to auto-remove that `cfg` that was causing problems, so that at least removes step 2 ("Delete existing globus cfg file") from the process.